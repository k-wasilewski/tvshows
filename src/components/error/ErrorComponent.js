import React from 'react';
import {styles} from "../../styles/styles";
import {useHistory} from 'react-router-dom';
import {ErrorMsg} from './ErrorMsg';
import Button from '@material-ui/core/Button';

export function ErrorComponent(props) {
    const history = useHistory();
    const classes = styles();

    return (
        <div className={classes.dashboard}>
            <ErrorMsg msg={props.msg} />
            <Button variant='outlined' size='small' onClick={() => history.push('/')}>
                Powr√≥t</Button>
        </div>
    );
};

export default ErrorComponent;
