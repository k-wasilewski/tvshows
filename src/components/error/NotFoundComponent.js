import React from 'react';
import {styles} from "../../styles/styles";
import {useHistory} from 'react-router-dom';
import {ErrorMsg} from './/ErrorMsg';
import Button from '@material-ui/core/Button';

export function NotFoundComponent(props) {
    const history = useHistory();
    const classes = styles();

    return (
        <div className={classes.dashboard}>
            <ErrorMsg msg={'Nie znaleziono strony'} />
            <Button variant='outlined' size='small' onClick={() => history.push('/')}>
                Powr√≥t</Button>
        </div>
    );
};

export default NotFoundComponent;
