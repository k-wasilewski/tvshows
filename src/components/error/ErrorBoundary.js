import React from "react";
import ErrorComponent from "./ErrorComponent";

export default class ErrorBoundary extends React.Component {
    state = {
        errorMessage: ''
    };

    static getDerivedStateFromError(error) {
        return {errorMessage: error.toString()};
    }

    componentDidCatch(error, info) {
        this.logErrorToServices(error.toString(), info.componentStack);
    }

    logErrorToServices = console.log;   //better logging

    render() {
        if (this.state.errorMessage) {
            return (
                <ErrorComponent msg='Wystąpił błąd w aplikacji' />
            );
        }
        return this.props.children;
    };
};