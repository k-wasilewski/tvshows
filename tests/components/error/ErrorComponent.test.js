import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorComponent} from "../../../src/components/error/ErrorComponent";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Button} from "@material-ui/core";
import {Router} from "react-router-dom";

describe("ErrorComponent rendering specification", () => {
    it('ErrorComponent is rendered', () => {
        const component = renderer.create(
            <ErrorComponent/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ErrorComponent functional specification", () => {
    it('button onClick redirects to landing page onClick', (done) => {
        configure({adapter: new Adapter()});

        const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

        const component = mount(
            <Router history={historyMock}>
                <ErrorComponent msg='Nie znaleziono strony' />
            </Router>
        );

        const returnBtn = component.find(Button).at(0);
        returnBtn.simulate('click');
        component.update();

        setTimeout(function () {
            expect(historyMock.push).toHaveBeenCalledWith('/');

            done();
        }, 500);
    });
});