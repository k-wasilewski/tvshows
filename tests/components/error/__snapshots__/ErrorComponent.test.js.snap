// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent rendering specification ErrorComponent is rendered 1`] = `
<div
  className="makeStyles-App-1"
>
  <div
    className="makeStyles-dashboard-12"
  >
    <span
      className="makeStyles-errorMsg-6"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Powr√≥t
      </span>
    </button>
  </div>
</div>
`;
